src/components/Verify.tsx

```diff
@@ -130,80 +135,91 @@ export const Verify = () => {
-                  <SelectChain
-                    value={creationTxHashes[index].chain}
-                    onChange={(chainValue) => handleChainChange(index, chainValue)}
-                    options={chains}
-                  />
+            {fields.map((txHash, index) => (
+              <>
+                <div key={index} className='flex items-center'>
+                  <div>
+                    <label className='text-sm leading-6 text-gray-400'>Chain</label>
+                    <SelectChain
+                      value={selectedChains[index]}
+                      onChange={(chainValue) => {
+                        setSelectedChains((prev) => {
+                          prev[index] = chainValue;
+                          return prev;
+                        });
+                        setValue(`creationTxHashes.${index}.chainId`, chainValue.id);
+                      }}
+                      options={chains}
+                    />
+                  </div>
+                  <div className='ml-2 flex-grow'>
+                    <label className='block text-sm leading-6 text-gray-400'>
+                      Transaction Hash<span className='text-red-600'> *</span>
+                    </label>
+                    <input
+                      placeholder='default'
+                      className='w-full rounded-md border-0 py-1.5 pl-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-900 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
+                      {...register(`creationTxHashes.${index}.hash` as const, {
+                        required: 'This is required',
+                      })}
+                    />
+                  </div>
+                  {selectedChains.length > 1 && (
+                    <GrFormClose
+                      className='cursor-pointer self-start'
+                      onClick={() => {
+                        remove(index);
+                        selectedChains.splice(index, 1);
+                        setSelectedChains(selectedChains);
+                      }}
+                    />
+                  )}
                 </div>
-                <div className='ml-2 flex-grow'>
-                  <label className='block text-sm leading-6 text-gray-400'>Transaction Hash</label>
-                  <input
-                    required
-                    onBlur={(e: React.ChangeEvent<HTMLInputElement>) =>
-                      handleTxHashInput(index, 'hash', e.target.value)
-                    }
-                    placeholder='default'
-                    className='w-full rounded-md border-0 py-1.5 pl-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
-                  />
+
+                <div className='mb-2 mt-1 flex'>
+                  <FormErrorMessage error={errors?.buildHint?.message} />
```

Comment:
A few items:
- Let's validate the tx hash is a hex string that starts with 0x and is the correct length
- If I click the verify button and have an empty tx hash field, no error message shows up under that field (it does nicely focus on the invalid field though)
- Similarly if I click the verify button and have two rows with the same chain that should error

Reply:
Validate tx hash, and make tx hash required. https://github.com/ScopeLift/cove-frontend/pull/9/commits/6132df5f1bf01d208352b7647dcb2a7bda18b25f

The same chain error should be handled in a modal or toast when the user tries to submit the form. It is pretty complicated to get everything working in the form itself and I think it would be a lower lift to throw an error on submit.